src/main.rs

```diff
@@ -195,167 +193,172 @@ fn get_setup(args: &[String]) -> Setup {
+            "INDEX",
         )
         .optopt(
             "",
             "initial-volume",
-            "Initial volume in %, once connected (must be from 0 to 100)",
+            "Initial volume in %, once connected {0..100}. Defaults to 50.",
             "VOLUME",
         )
         .optopt(
             "",
             "zeroconf-port",
             "The port the internal server advertised over zeroconf uses.",
-            "ZEROCONF_PORT",
+            "PORT",
         )
         .optflag(
             "",
             "enable-volume-normalisation",
-            "Play all tracks at the same volume",
+            "Play all tracks at the same volume.",
         )
         .optopt(
             "",
             "normalisation-method",
-            "Specify the normalisation method to use - [basic, dynamic]. Default is dynamic.",
-            "NORMALISATION_METHOD",
+            "Specify the normalisation method to use {basic|dynamic}. Defaults to dynamic.",
+            "METHOD",
         )
         .optopt(
             "",
             "normalisation-gain-type",
-            "Specify the normalisation gain type to use - [track, album]. Default is album.",
-            "GAIN_TYPE",
+            "Specify the normalisation gain type to use {track|album}. Defaults to album.",
+            "TYPE",
         )
         .optopt(
             "",
             "normalisation-pregain",
-            "Pregain (dB) applied by volume normalisation",
+            "Pregain (dB) applied by volume normalisation. Defaults to 0.",
             "PREGAIN",
         )
         .optopt(
             "",
             "normalisation-threshold",
-            "Threshold (dBFS) to prevent clipping. Default is -1.0.",
+            "Threshold (dBFS) to prevent clipping. Defaults to -1.0.",
             "THRESHOLD",
         )
         .optopt(
             "",
             "normalisation-attack",
-            "Attack time (ms) in which the dynamic limiter is reducing gain. Default is 5.",
-            "ATTACK",
+            "Attack time (ms) in which the dynamic limiter is reducing gain. Defaults to 5.",
+            "TIME",
         )
         .optopt(
             "",
             "normalisation-release",
-            "Release or decay time (ms) in which the dynamic limiter is restoring gain. Default is 100.",
-            "RELEASE",
+            "Release or decay time (ms) in which the dynamic limiter is restoring gain. Defaults to 100.",
+            "TIME",
         )
         .optopt(
             "",
             "normalisation-knee",
-            "Knee steepness of the dynamic limiter. Default is 1.0.",
+            "Knee steepness of the dynamic limiter. Defaults to 1.0.",
             "KNEE",
         )
         .optopt(
             "",
             "volume-ctrl",
-            "Volume control type - [linear, log, fixed]. Default is logarithmic",
-            "VOLUME_CTRL"
+            "Volume control type {cubic|fixed|linear|log|none}. Defaults to log.",
```

Comment:
"none" is a spurious leftover, I've yet to map this to "fixed" with initial volume 100.

Reply:
I don't think that "none" should be an option, "fixed" is already a thing and it's not a complicated concept. It seems extremely redundant to me. I honestly don't know why initial volume doesn't already default to 100? I'm also not sure why cubic is a thing if log is basically a superior version of it?

Reply:
> I don't think that "none" should be an option, "fixed" is already a thing and it's not a complicated concept. It seems extremely redundant to me.

You probably have a point there. I'll drop it unless there's some vocal majority. Doing the mapping in `main.rs` wouldn't be very hard, however less code transparent than just doing `VolumeCtrl::find`.

> I honestly don't know why initial volume doesn't already default to 100?

If I had to take a guess, it would be to assume a safe playback level as opposed to blasting out your drivers or ears.

> I'm also not sure why cubic is a thing if log is basically a superior version of it?

One point you could make is that cubic offers more granular control in the upper volume range. The other is not to scrap something that is already being used.

Reply:
> If I had to take a guess, it would be to assume a safe playback level as opposed to blasting out your drivers or ears.

If possible librespot should query the volume of the device that initiated the connection and set it's volume to that IMHO. Alternatively in the case of hardware volume and being a part of a system like Moode it should respect the existing volume so that when a user "changes sources"  the volume stays the same. IMHO librespot shouldn't just trample the volume level arbitrarily.

Reply:
> If possible librespot should query the volume of the device that initiated the connection and set it's volume to that IMHO.

I don't think that is how Connect works. When you connect to an official playback device with an official client, the client takes on the volume of the playback device, not the other way around.

> Alternatively in the case of hardware volume and/or being a part of a system like Moode it should respect the existing volume so that when a user "changes sources" the volume stays the same. IMHO librespot shouldn't just trample the volume level arbitrarily.

I agree and while I looked into this briefly, but it's not so trivial to do because of the decoupling between the `connect` and `playback` crates. I have to think about this further. (Maybe introduce something like `task.get_volume().unwrap_or_else(|| initial_volume)` in `spirc` -- suggestions anyone?)

Reply:
> I don't think that is how Connect works. When you connect to an official playback device with an official client, the client takes on the volume of the playback device, not the other way around.

Yep, that does seem to be the case at least between my phone and the desktop app. It still seems counteractive to me.

> suggestions anyone?

I guess in case of something like a audio appliance distro like moode really you could just set the volume to fixed 100 and moode could handle volume downstream.

Reply:
> > I don't think that is how Connect works. When you connect to an official playback device with an official client, the client takes on the volume of the playback device, not the other way around.
> 
> Yep, that does seem to be the case at least between my phone and the desktop app. It still seems counterintuitive to me.

I'd say it makes sense. After all 80% on your phone isn't quite the same dB SPL as 80% on your 100W stereo 😆 

> > suggestions anyone?
> 
> I guess in case of something like a audio appliance distro like moode really you could just set the volume to fixed 100 and moode could handle volume downstream.

I contributed a thing or two to moOde and can say that's not how moOde works today. It could, it it would use the Alsa mixer everywhere, but it doesn't. It's mostly a front-end to `librespot`, `shairpoint` and `mpd` for streaming radio and local storage. (I don't mean this as derogatory in any way, I love moOde, this is just explaining the architecture.)

Reply:
> I'd say it makes sense. After all 80% on your phone isn't quite the same dB SPL as 80% on your 100W stereo

I know but it's a little strange to start with your phone at whatever volume level and then connect to a device and it changes your phone's volume. To me that not an expected behavior.

> I contributed a thing or two to moOde and can say that's not how moOde works today. It could, it it would use the Alsa mixer everywhere, but it doesn't. It's mostly a front-end to librespot, shairpoint and mpd for streaming radio and local storage. (I don't mean this as derogatory in any way, I love moOde, this is just explaining the architecture.)

All things being equal you'd think you want a common consistent volume control between services. IMHO I would expect all services to be the same volume. You might still provide individual volume control but more for level matching.

Reply:
I really don't have a dog in the fight though. My argument is mostly academic as I generally just set librespot to fixed 100 and use the volume control on my stereo receiver.

Reply:
> > I'd say it makes sense. After all 80% on your phone isn't quite the same dB SPL as 80% on your 100W stereo
> 
> I know but it's a little strange to start with your phone at whatever volume level and then connect to a device and it changes your phone's volume. To me that's not an expected behavior.

Are you sure that's what happens? Not on my iPhone. Say my phone volume is set at 80% and the Connect device at 50%. When I connect, the volume slider in Spotify is now set at 50%. Open some other app and the phone volume is again 80%.

Regardless, as much as I enjoy an academic discussion this is just how it is.
 
> > I contributed a thing or two to moOde and can say that's not how moOde works today. It could, it it would use the Alsa mixer everywhere, but it doesn't. It's mostly a front-end to librespot, shairpoint and mpd for streaming radio and local storage. (I don't mean this as derogatory in any way, I love moOde, this is just explaining the architecture.)
> 
> All things being equal you'd think you'd want a common consistent volume control between services. IMHO I would expect all services to be the same volume. You might still provide individual volume control but more for level matching.

Clearly out of scope for this PR :wink:

Reply:
> Clearly out of scope for this PR

Yep. That would be something you'd take up with moode.

Reply:
Removed in 11f5f90.